src/equations/ideal_glm_mhd_3d.jl

```diff
@@ -67,19 +68,20 @@ function initial_condition_convergence_test(x, t, equations::IdealGlmMhdEquation
     # Alfvén wave in three space dimensions
     # Altmann thesis http://dx.doi.org/10.18419/opus-3895
     # domain must be set to [-1, 1]^3, γ = 5/3
+    RealT = eltype(x)
     p = 1
-    omega = 2 * pi # may be multiplied by frequency
+    omega = 2 * convert(RealT, pi) # may be multiplied by frequency
     # r: length-variable = length of computational domain
-    r = 2
+    r = 2.0f0
```

Comment:
```suggestion
    r = convert(RealT, 2)
```

Stuff like `1 / sqrt(r^2 + 1)` is computed below. Setting `r` to `Float32` changes the result to `Float32` in all cases. That can cause some of the CI errors

Reply:
But `e = convert(RealT, 0.2)` and `SVector(...)` in return have already ensured the type would not be `Float32` in all cases - CI tests for type stability have all been passed.

Reply:
What we have done to integers before is to make it into `Float32` please see https://github.com/trixi-framework/Trixi.jl/blob/29eb55a05964964187c30475f3ccc578423a48cd/src/equations/compressible_euler_1d.jl#L70-L85

Reply:
@ranocha

Reply:
There will be no difference for the case with `L` above since `1.0f0 / 2` can be represented exactly as `Float32`. However, it makes a difference in the present case:

```julia
julia> r = 2.0; 1 / sqrt(r^2 + 1)
0.4472135954999579

julia> r = 2.0f0; 1 / sqrt(r^2 + 1)
0.4472136f0
```

Reply:
That doesn't affect type stability since the `Float32` is converted later, but it changes some intermediate results - and thus also the final result

Reply:
Oh I see thanks!!

Reply:
So the failed CI tests mainly run for the precision, right?

Reply:
Another question - why not directly code as `f=0.5f0` (and many other places like this)? Is that for making the function more clear and closer to the original formula?

Reply:
> Another question - why not directly code as `f=0.5f0` (and many other places like this)? Is that for making the function more clear and closer to the original formula?

Yes